src/io/unix/sync.rs

```diff
@@ -59,39 +59,50 @@ impl File {
         self.size.load(Ordering::SeqCst)
     }
 
-    pub(crate) async fn write_append_all(&self, buf: Bytes) -> IOResult<()> {
-        let offset = self.size.fetch_add(buf.len() as u64, Ordering::SeqCst);
-        self.write_all_at(offset, buf).await
-    }
-
-    pub(crate) async fn write_append_all_buffers(
+    pub(crate) async fn write_append_writable_data<R: Send + 'static>(
         &self,
-        first_buf: Bytes,
-        second_buf: Bytes,
-    ) -> IOResult<()> {
-        let first_len = first_buf.len() as u64;
-        let second_len = second_buf.len() as u64;
-        let len = first_len + second_len;
+        c: impl WritableDataCreator<R>,
+        len: u64,
+    ) -> Result<R> {
         let mut offset = self.size.fetch_add(len, Ordering::SeqCst);
```

Comment:
The `size` update must be inside the block of code passed to `inplace_sync_call` or `background_sync_call`. This is because there is an `await` below that can be the point of cancellation